pact/marmalade.repl

```diff
@@ -580,9 +584,124 @@
 
 (commit-tx)
 
+(begin-tx "sale fixed-quote-policy failure")
+(env-hash (hash "fixed-quote-sale-tx2"))
+(use marmalade.ledger)
+(env-chain-data {"block-height": 100})
+(env-sigs
+  [{'key:'bob
+   ,'caps:
+    [(marmalade.ledger.OFFER "project-2" "bob" 0.2 120)]}
+ ])
+
+(env-data
+ {
+  "quote": {
+      "fungible": coin
+     ,"price": 1.0
+     ,"recipient": "alice"
+     ,"recipient-guard": {"keys": ["alice"], "pred":"="}
+   }
+ })
+
+(expect-failure
+  "Sale fails with non-existent recipient account"
+  "Failure: Tx Failed: with-read: row not found: alice"
+  (sale 'project-2 'bob 0.2 120))
+
+(env-data
+  {
+   "quote": {
+       "fungible": coin
+      ,"price": 1.0
+      ,"recipient": "alice"
+      ,"recipient-guard": {"keys": ["wrong-alice"], "pred":"="}
+    }
+  ,"alice": {"keys": ["alice"], "pred":"="}
+ })
+
+(expect
+  "Create recipient account alice"
+  "Write succeeded"
+  (coin.create-account 'alice (read-keyset 'alice)))
+
+(expect-failure
+  "Sale fails with wrong recipient guard"
+  "Failure: Tx Failed: Recipient guard does not match"
+  (sale 'project-2 'bob 0.2 120))
+
+(env-data
+  {
+   "quote": {
+       "fungible": ns
+      ,"price": 1.0
+      ,"recipient": "alice"
+      ,"recipient-guard": {"keys": ["alice"], "pred":"="}
+    }
+  })
+
+(expect-failure
+  "Sale fails with non-fungible in quote"
+  "fungible: Failure: Type error: expected module{fungible-v2}, found module{}"
+  (sale 'project-2 'bob 0.2 120))
+
+(env-data
+  {
+   "quote": {
+       "fungible": coin
+      ,"price": -1.0
+      ,"recipient": "alice"
+      ,"recipient-guard": {"keys": ["alice"], "pred":"="}
+    }
+  })
+
+(expect-failure
+  "Sale fails with negative price in quote"
+  "Offer amount must be positive"
+  (sale 'project-2 'bob 0.2 120))
+
+(env-data
+  {
+   "quote": {
+       "fungible": coin
+      ,"price": 1.0
+      ,"recipient": {"account": "alice"}
+      ,"recipient-guard":  {"keys": ["alice"], "pred":"="}
+    }
+  })
+
+(expect-failure
+  "Sale fails with invalid recipient in quote"
+  "Type error: expected string, found object:*"
```

Comment:
I think this test would be better with an invalid account as opposed to an invalid type. Stuff that gets enforced by pact itself doesn't need tests. Also since the below tests the guard, that shows the lookup is being performed, so you probably can just remove this test altogether and have the same coverage (since valid account enforcement is not the responsibility of the module you're testing)

Reply:
Also removed the case with invalid guard type for the same reason.